ambry-commons/src/main/java/com/github/ambry/compression/CompressionMap.java

```diff
@@ -53,19 +54,29 @@ public void add(Compression compressionInstance) {
 
   /**
    * Find the name of the algorithm used to compress the specified compressed buffer.
+   * This method does not alter the index in compressedBuffer.
    *
    * @param compressedBuffer The compressed buffer generated by the compress() method.
-   * @param compressedBufferOffset Offset in compressedBuffer to start reading.
-   * @param compressedDataSize The size of compressed data in compressedBuffer.
    * @return Name of the compression algorithm.
    */
-  public String getAlgorithmName(byte[] compressedBuffer, int compressedBufferOffset, int compressedDataSize) {
-    BaseCompression.verifyCompressedBuffer(compressedBuffer, compressedBufferOffset, compressedDataSize);
+  public String getAlgorithmName(ByteBuffer compressedBuffer) {
+    Utils.checkNotNullOrEmpty(compressedBuffer, "compressedBuffer cannot be null or empty.");
 
-    // The first byte is the version. Skip the version to reach the algorithm name.
-    int algorithmNameOffset = compressedBufferOffset + 1;
-    byte algorithmNameLength = compressedBuffer[algorithmNameOffset];
-    return new String(compressedBuffer, algorithmNameOffset + 1, algorithmNameLength, StandardCharsets.UTF_8);
+    // The first byte is the version. Skip the version and read the algorithm name length.
+    compressedBuffer.mark();
+    try {
+      compressedBuffer.get();  // Skip the version.
+      byte algorithmNameLength = compressedBuffer.get();
```

Comment:
if u know the first byte is to be skipped, then use buffer.get(8) ?

Reply:
i get that using relative get is convenient to write but makes for a more verbose code with lines that don't do anything

Reply:
you could have constants like ALGO_LEN_START_IDX, ALGO_NAME_START_IDX ?